### PR TITLE
Sdks-3619_Flutter-to_do_app_update

### PR DESCRIPTION
**Pull request by george-bafaloukas-forgerock**
Updating and moving into the repo the Flutter sample application.
At the moment this is the `complete` project.
Start version with TODOs to follow.
This contains the DART flutter code, iOS Bridge code and Android Code

### CODE REVIEW

File: flutter/flutter_todo/android/app/src/main/java/com/example/flutter_todo/FRAuthSampleBridge.java [hunk 1]
```diff
@@ -0,0 +1,518 @@
+package com.example.flutter_todo;
+
+/*
+ * Copyright (c) 2022-2024 Ping. All rights reserved.
```
Comment (rodrigoareis) [inline flutter/flutter_todo/android/app/src/main/java/com/example/flutter_todo/FRAuthSampleBridge.java:4]: Copyright (c) 2022-2024 Ping Identity. All rights reserved.

File: flutter/flutter_todo/android/app/src/main/kotlin/com/example/flutter_todo/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,44 @@
+package com.example.flutter_todo
```
Comment (rodrigoareis) [inline flutter/flutter_todo/android/app/src/main/kotlin/com/example/flutter_todo/MainActivity.kt:1]: Missing copyright

File: flutter/flutter_todo/ios/Runner/AppDelegate.swift [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import Flutter
```
Comment (rodrigoareis) [inline flutter/flutter_todo/ios/Runner/AppDelegate.swift:1]: Missing copyright

File: flutter/flutter_todo/ios/Runner/FRAuthSampleBridge.swift [hunk 4]
```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright (c) 2022 ForgeRock. All rights reserved.
```
Comment (rodrigoareis) [inline flutter/flutter_todo/ios/Runner/FRAuthSampleBridge.swift:2]: Update copyright

File: flutter/flutter_todo/ios/Runner/FRAuthSampleHelpers.swift [hunk 5]
```diff
@@ -0,0 +1,49 @@
+//
+//  FRAuthSampleHelpers.swift
```
Comment (rodrigoareis) [inline flutter/flutter_todo/ios/Runner/FRAuthSampleHelpers.swift:2]: Update copyright

File: flutter/flutter_todo/ios/Runner/FRAuthSampleStructs.swift [hunk 6]
```diff
@@ -0,0 +1,227 @@
+//
+//  FRAuthSampleStructs.swift
```
Comment (rodrigoareis) [inline flutter/flutter_todo/ios/Runner/FRAuthSampleStructs.swift:2]: Update copyright

File: flutter/flutter_todo/ios/Runner/FRAuthSampleStructs.swift [hunk 7]
```diff
@@ -0,0 +1,227 @@
...
+//  Runner
+//
+//  Created by George Bafaloukas on 03/03/2022.
+//
+
+import Foundation
+import FRCore
+import FRAuth
+
+public struct FRNode: Encodable {
```
Comment (rodrigoareis) [inline flutter/flutter_todo/ios/Runner/FRAuthSampleStructs.swift:12]: Add document

File: flutter/flutter_todo/ios/Runner/FRAuthSampleStructs.swift [hunk 8]
```diff
@@ -0,0 +1,227 @@
...
+    response["authServiceId"] = self.authServiceId
+    response["stage"] = self.stage
+    response["description"] = self.pageDescription
+    response["header"] = self.pageHeader
+    response["callbacks"] = self.callbacks
+    return try response.toJson()
+  }
+}
+
+public struct FRCallback: Encodable {
```
Comment (rodrigoareis) [inline flutter/flutter_todo/ios/Runner/FRAuthSampleStructs.swift:59]: Add document

File: flutter/flutter_todo/ios/Runner/Runner-Bridging-Header.h [hunk 9]
```diff
@@ -0,0 +1 @@
+#import "GeneratedPluginRegistrant.h"
```
Comment (rodrigoareis) [inline flutter/flutter_todo/ios/Runner/Runner-Bridging-Header.h:1]: Missing copyright

File: flutter/flutter_todo/ios/RunnerTests/RunnerTests.swift [hunk 10]
```diff
@@ -0,0 +1,12 @@
+import Flutter
```
Comment (rodrigoareis) [inline flutter/flutter_todo/ios/RunnerTests/RunnerTests.swift:1]: Missing copyright

File: flutter/flutter_todo/lib/FRCallback.dart [hunk 11]
```diff
@@ -0,0 +1,61 @@
+//  FRCallback.dart
+//
+//  Copyright (c) 2022 ForgeRock. All rights reserved.
```
Comment (rodrigoareis) [inline flutter/flutter_todo/lib/FRCallback.dart:3]: Update copyright

File: flutter/flutter_todo/lib/FRNode.dart [hunk 12]
```diff
@@ -0,0 +1,50 @@
+//  FRNode.dart
+//
+//  Copyright (c) 2022 ForgeRock. All rights reserved.
```
Comment (rodrigoareis) [inline flutter/flutter_todo/lib/FRNode.dart:3]: Update copyright

File: flutter/flutter_todo/lib/FRNode.dart [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+//  of the MIT license. See the LICENSE file for details.
+
+/*
+The FRNode class is native Dart class replicating the structure of the SDKs Node objects.
+This is used in order to natively parse and encode/decode the JSON objects returned from the SDKs.
+*/
+
+import 'FRCallback.dart';
+
+class FRNode {
```
Comment (rodrigoareis) [inline flutter/flutter_todo/lib/FRNode.dart:15]: Add document

File: flutter/flutter_todo/lib/main.dart [hunk 14]
```diff
@@ -0,0 +1,54 @@
+import 'package:flutter/material.dart';
```
Comment (rodrigoareis) [inline flutter/flutter_todo/lib/main.dart:1]: Missing copyright

File: flutter/flutter_todo/lib/register.dart [hunk 15]
```diff
@@ -0,0 +1,315 @@
+import 'dart:convert';
```
Comment (rodrigoareis) [inline flutter/flutter_todo/lib/register.dart:1]: Missing copyright

File: flutter/flutter_todo/lib/todolist.dart [hunk 16]
```diff
@@ -0,0 +1,363 @@
+import 'dart:convert';
```
Comment (rodrigoareis) [inline flutter/flutter_todo/lib/todolist.dart:1]: Missing copyright

File: flutter/flutter_todo/macos/Runner/AppDelegate.swift [hunk 17]
```diff
@@ -0,0 +1,9 @@
+import Cocoa
```
Comment (rodrigoareis) [inline flutter/flutter_todo/macos/Runner/AppDelegate.swift:1]: Missing copyright

File: flutter/flutter_todo/macos/Runner/MainFlutterWindow.swift [hunk 18]
```diff
@@ -0,0 +1,15 @@
+import Cocoa
```
Comment (rodrigoareis) [inline flutter/flutter_todo/macos/Runner/MainFlutterWindow.swift:1]: Missing copyright

File: flutter/flutter_todo/macos/RunnerTests/RunnerTests.swift [hunk 19]
```diff
@@ -0,0 +1,12 @@
+import Cocoa
```
Comment (rodrigoareis) [inline flutter/flutter_todo/macos/RunnerTests/RunnerTests.swift:1]: Missing copyright

**Review by rodrigoareis — CHANGES_REQUESTED** [review]
Overall changes looks good, left minor comments

File: flutter/flutter_todo/macos/Runner/AppDelegate.swift (hunk 17 above)
Comment (george-bafaloukas-forgerock) [reply flutter/flutter_todo/macos/Runner/AppDelegate.swift:1]: Will remove

File: flutter/flutter_todo/macos/Runner/MainFlutterWindow.swift (hunk 18 above)
Comment (george-bafaloukas-forgerock) [reply flutter/flutter_todo/macos/Runner/MainFlutterWindow.swift:1]: Will remove

File: flutter/flutter_todo/macos/RunnerTests/RunnerTests.swift (hunk 19 above)
Comment (george-bafaloukas-forgerock) [reply flutter/flutter_todo/macos/RunnerTests/RunnerTests.swift:1]: Will remove

File: flutter/flutter_todo/lib/FRNode.dart (hunk 13 above)
Comment (george-bafaloukas-forgerock) [reply flutter/flutter_todo/lib/FRNode.dart:17]: Done

File: flutter/flutter_todo/ios/Runner/FRAuthSampleStructs.swift (hunk 8 above)
Comment (george-bafaloukas-forgerock) [reply flutter/flutter_todo/ios/Runner/FRAuthSampleStructs.swift]: Done

File: flutter/flutter_todo/ios/Runner/FRAuthSampleStructs.swift (hunk 7 above)
Comment (george-bafaloukas-forgerock) [reply flutter/flutter_todo/ios/Runner/FRAuthSampleStructs.swift:15]: Done

File: iOS/swiftui-davinci/Davinci/Davinci/DavinciViewModel.swift [hunk 20]
```diff
@@ -21,15 +21,15 @@ import PingOrchestrate
...
-public let davinci = DaVinci.createDaVinci { config in
-  //TODO: Provide here the Server configuration. Add the PingOne server Discovery Endpoint and the OAuth2.0 client details
-  config.module(OidcModule.config) { oidcValue in
-    oidcValue.clientId = <#"Client ID"#>
-    oidcValue.scopes = [<#"scope1"#>, <#"scope2"#>, <#"scope3"#>]
-    oidcValue.redirectUri = <#"Redirect URI"#>
-    oidcValue.discoveryEndpoint = <#"Discovery Endpoint"#>
-  }
-}
+    public let davinci = DaVinci.createDaVinci { config in
```
Comment (rodrigoareis) [inline iOS/swiftui-davinci/Davinci/Davinci/DavinciViewModel.swift:24]: Revert this change

**Review by rodrigoareis — APPROVED** [review]
Looks good

**Review by jeyanthanperiyasamy — APPROVED** [review]

File: flutter/flutter_todo/android/app/src/main/java/com/example/flutter_todo/FRAuthSampleBridge.java [hunk 21]
```diff
@@ -0,0 +1,607 @@
+/*
```
Comment (jeyanthanperiyasamy) [inline flutter/flutter_todo/android/app/src/main/java/com/example/flutter_todo/FRAuthSampleBridge.java:1]: in future , please change this to kotlin, because we are going to use lot of suspend methods. that would be tricky to call from java

Comment (george-bafaloukas-forgerock) [reply flutter/flutter_todo/android/app/src/main/java/com/example/flutter_todo/FRAuthSampleBridge.java:1]: yeah when we re-write that we should do Kotlin. not a priority at the moment though
